core/app/models/spree/shipment.rb

```diff
@@ -14,6 +21,10 @@ class Shipment < Spree::Base
     has_many :state_changes, as: :stateful
     has_many :cartons, -> { uniq }, through: :inventory_units
 
+    validates :state, presence: true, inclusion: { in: [READY, PENDING, SHIPPED, CANCELED] }
```

Comment:
`presence: true` is superfluous if we also check the inclusion in something that's not having `nil`.

Maybe we could also add default value for the `state` column?